chore: Add FOSSA check to GH actions

ghislainbourgeois opened the pull request.

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.14.4'
```
```suggestion
        go-version-file: 'go.mod'
```

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
```suggestion
    - uses: actions/checkout@v4
```

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
```
```suggestion
      uses: actions/setup-go@v5
```

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.14.4'
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v ./...
```
Should not this be deleted given that you already have a `build` and `unit-tests` actions below?

ghislainbourgeois replied:
I could swear I had removed those, but seems not! I deleted that duplicate section.

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+          go-version-file: 'go.mod'
+
+      - name: Build
+        run: go build
```
```suggestion
        run: go build ./...
```

ghislainbourgeois replied:
Done

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+      - name: Build
+        run: go build
+
+  license-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: reuse lint
+        uses: fsfe/reuse-action@v2
+
+  fossa-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: FOSSA scan
+        uses: fossa-contrib/fossa-action@v3
+        with:
+          fossa-api-key: 9dc8fa92e3dd565687317beb87b56d89
```
Should we move these actions to the end of the file such we try to keep as much consistency as possible with the GHA file in the other repos?

ghislainbourgeois replied:
Done

gab-arrobo commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,71 @@
+name: Main workflow
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
To make it similar to the GHA file in the other repos
```suggestion
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```

ghislainbourgeois replied:
Done

gab-arrobo reviewed (approved):
+1

gab-arrobo commented:
@ghislainbourgeois there is license issue with one file (Namf_Communication/.openapi-generator/VERSION). How should we fixed it?

ghislainbourgeois commented:
> @ghislainbourgeois there is license issue with one file (Namf_Communication/.openapi-generator/VERSION). How should we fixed it?

I removed the file, as it seems unused and it is only present in that specific directory.

gab-arrobo reviewed (approved):
+1
